docs/design/drf - fairshare.md

```diff
@@ -0,0 +1,179 @@
+# Namespace fair share
+
+[@lminzhw](http://github.com/lminzhw); May 8, 2019
+
+## Motivation
+
+`Queue` was introduced in [kube-batch](http://github.com/kubernetes-sigs/kube-batch) to share resources among users.
+
+But, the user in the same `Queue` are equivalent during scheduling. For example, we have a `Queue` contains a small amount of resources, and there are 10 pods belong to UserA and 1000 pods belong to UserB. In this case, pods of UserA would have less probability to bind with node.
+
+So, we need a more fine-grained strategy to balance resource usage among users in the same `Queue`.
+
+In consideration of multi-user model in kubernetes, we use namespace to distinguish different user. Each namespace would have its weight to control resources usage.
+
+## Function Specification
+
+Weight have these features:
+> 1. `Queue` level
+> 2. an `integer` with default value 1
+> 3. record in namespace `quota`
+> 4. higher value means more resources after balancing
+
+### where is the weight
+
+```yaml
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  namespace: default
+spec:
+  hard:
+    limits.memory: 2Gi
+    volcano.sh/namespace.weight: 1  <- this field represent the weight of this namespace
+```
+
+If many `ResourceQuota` in the same namespace have weight, the weight for this namespace is the highest one of them.
+
+This weight should be positive, any invalid value is treated as default value 1.
+
+### Scheduler Framework
+
+Introduce two new fields in SchedulerCache
+
+```go
+type NamespaceInfo struct {
+    Weight int
+}
+
+type SchedulerCache struct {
+    ...
+    quotaInformer    infov1.ResourceQuotaInformer
+    ...
+    NamespaceInfo  map[string]*kbapi.NamespaceInfo
+    ...
+}
+```
+
+The Scheduler will watch the lifecycle of `ResourceQuota` by `quotaInformer`, and refresh the info in `NamespaceInfo`.
+
+In `openSession` function, we should pass the `NamespaceInfo` through function `cache.Snapshot` into `Session` by using a new filed in `Session`/`ClusterInfo` struct.
+
+```go
+type Session struct {
+    ...
+    NamespaceInfo  map[string]*kbapi.NamespaceInfo
```

Comment:
Prefer to `UserInfo`, similar to other place.

Reply:
> Prefer to `UserInfo`, similar to other place.

Related: https://github.com/volcano-sh/kube-batch/pull/14#discussion_r285819379